Add data preprocessing and Gemini API integration setup

vincentkyalomusembi opened the pull request:
- Added requirements.txt with necessary dependencies
- Set up branch for data preprocessing functions
- Ready to add data loading and preprocessing code

vincentkyalomusembi committed:
data preprocessing & gemini API intergration

vincentkyalomusembi committed:
Add data preprocessing functions

- Add data_loader.py with Fashion-MNIST loading functions
- Add preprocessing.py for MLP and CNN data preparation

vincentkyalomusembi committed:
Add data preprocessing with Keras Fashion-MNIST

- Use Keras built-in Fashion-MNIST dataset (no large CSV files)
- Add data_loader.py with Fashion-MNIST loading functions
- Add preprocessing.py for MLP and CNN data preparation
- Add gemini_integration.py for price/store API calls
- Add test_preprocessing.py to verify functions work

Lenida-Daniels commented on test_preprocessing.py:
```diff
@@ -0,0 +1,72 @@
+    """Test class names function"""
+    print("\nTesting class names...")
+    try:
+        classes = get_class_names()
```
'get_class_names' is not defined

vincentkyalomusembi replied:
i will work on it

vincentkyalomusembi committed:
Update README with comprehensive project description

- Add detailed project overview and features
- Document technical implementation and structure
- Include installation and usage instructions
- List real-world applications and team contributions
